### PR TITLE
Added tests for Button kind events and icon colors

### PR DESCRIPTION
**Pull request by ShimiSun**
#### What does this PR do?
Added tests for Button kind events and icon colors

#### Where should the reviewer start?
test file
#### What testing has been done on this PR?
Jest
#### How should this be manually tested?
Jest
#### Any background context you want to provide?
image

image

Updated:
image

#### What are the relevant issues?
#4539 
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
b/c

### CODE REVIEW

File: src/js/components/Button/__tests__/Button-kind-test.js [hunk 1]
```diff
@@ -37,6 +39,46 @@ describe('Button kind', () => {
...
+        theme={{
+          button: {
+            default: {
+              color: '#FFF',
+              background: '#000',
+            },
+          },
+        }}
+      >
+        <Button label="label" icon={<Add />} hoverIndicator />
```
Comment (taysea) [inline src/js/components/Button/__tests__/Button-kind-test.js:54]: The docs note that `hoverIndicator` should only be used with `plain` buttons, but this isn't a plain button. Do we still want to have a test case?

Comment (taysea) [reply src/js/components/Button/__tests__/Button-kind-test.js:54]: also if this test just has a single expect, i'm not sure we are capturing the `hoverIndicator` in the snapshot

Comment (ShimiSun) [reply src/js/components/Button/__tests__/Button-kind-test.js:54]: Good catch, the `hoverIndicator` slipped in and is not needed for this test, let me clean it up.

**Comment by ShimiSun** [comment]
Cleaned up the tests and improved coverage:

**Review by taysea — APPROVED** [review]
LGTM

**Pull request merged**